packages/ses/test/package/test.cjs

```diff
@@ -1,7 +1,13 @@
 // node test.cjs
 
+// this is a polyfill for node v10 and earlier
+if (!global.globalThis) {
+  global.globalThis = global;
+}
```

Comment:
@erights Is this a normalization we should subsume into the SES shim?

Reply:
Kinda. The problem is that as written it assumes that `global` is already bound to the global object, which is not the case on many JS platforms. Somewhere I saw a comprehensive shim for `globalThis` and it was hellacious. Since the SES shim gets to use the original unsafe `Function` constructor internally, how about something like:

```js
if (typeof globalThis === 'undefined') {
   const global = new UnsafeFunction('this')();
   global.globalThis = global;
}
```

Until then, if we continue to have code like that above in the test, we should still not have it unconditionally assume that `global` is a correct name for the global object. How about

```js
if (typeof globalThis === 'undefined' && typeof global === 'object') {
   global.globalThis = global;
}
```

Reply:
Does it make sense to leave this “shim” here since it is necessary for our tests to pass on Node.js 0.10 and we only run these tests under Node.js?

@kumavis I believe my preference would be to use the shim code @erights recommends inline here since it would ensure future portability of the test itself.

Reply:
We should only accept this shim if we consider Node 10 a platform we support. Since these are the only two known problems, this seems a reasonable goal, so I'm in favor. But in that case, we should also subsume one of these `globalThis` fixes into SES itself. I'm fine with this PR not doing that, and merely changing the test to use my second suggestion. But once merged we should take that second step, and remember to remove this code from the test.

Reply:
Ah, that being the case, I don’t think it’s too much to ask @kumavis to remove this shim and add your recommendation in SES proper so we don’t have to track future work.

Let’s put the shim in `src/lockdown-shim.js` at the top level (before repair, before lockdown). This is not a SES shim, but rather just an environment normalization we’ll do gratis.

Reply:
> Does it make sense to leave this “shim” here since it is necessary for our tests to pass on Node.js 0.10 and we only run these tests under Node.js?

FWIW, keep in mind v0.10 !== v10

Reply:
Gah. I meant v10. Thank you. I lived too long in Node v0.10.36 and it’s in muscle memory.

Reply:
I’ll take this as-is and we can, if we choose, subsume it later.

Reply:
https://github.com/Agoric/SES-shim/issues/479

Reply:
Hi @kumavis 
The objection to this code stands. As written it is not portable to non-node environments. At least change to my second code snippet above, as that will at least not fail directly in environments that don't have a `global`.

Unresolving